src/main/java/com/exactpro/epfast/decoder/message/DecoderState.java

```diff
@@ -50,6 +51,8 @@
 
     public final UnionRegister register = new UnionRegister();
 
+    public boolean valuePresent = true;
```

Comment:
I don't like it. We already have instruction pointer. What we need is to create conditional operator which executes different code paths depending on value read from presence map.